### DISCUSSION TITLE
tflite digit model with black & white images

### DISCUSSION THREAD

**Discussion by nliaudat** [jomjol/AI-on-the-edge-device#1780] [Ideas]
I mean retraining the digit model with  B/W only images could improve overall perfomances. 

The esp32 should also take PIXFORMAT_GRAYSCALE image and then convert it to B/W before processing.

Sample code available at :https://www.hackster.io/news/handwritten-digit-classification-with-arduino-and-microml-4d7d4ffec6b7

What do you think @haverland ?

**Comment by ghost** [comment]
Often discussed, but isn't.  Why? Only the first layer will be changed:

Grayscale model Parameters: (https://github.com/haverland/Tenth-of-step-of-a-meter-digit/blob/dev/dig-class100-s2_bw.ipynb )
Total params: 213,416
Trainable params: 213,094
Non-trainable params: 322

RGB model Parameters: ( https://github.com/haverland/Tenth-of-step-of-a-meter-digit/blob/dev/dig-class100-s2.ipynb)
Total params: 214,000
Trainable params: 213,674
Non-trainable params: 326

**Reply by nliaudat** [reply]
[CNN networks are not grayscale friends](https://towardsdatascience.com/transfer-learning-on-greyscale-images-how-to-fine-tune-pretrained-models-on-black-and-white-9a5150755c7a) .... 

I've seen you tried inference_input_type and inference_output_type attributes in b2n/models/prune_quantize.py
```
    #converter2.inference_input_type = tf.uint8
    #converter2.inference_output_type = tf.uint8
```
but it's commented out. Do you have bad results ?
